### PR TITLE
txn: add table name in dup entry error message

### LINKED ISSUE #38242: Include table name in the error message of 1062: duplicate entry
## Enhancement

Currently the error message looks like

```
[kv:1062]Duplicate entry '1' for key 'PRIMARY'
```

It can be hard to tell the exact column in the error message if multiple table has columns with same names, especially when `@@tidb_constraint_check_in_place_pessimistic = off`.

We can print the table name with the column name to make it clearer, like

```
[kv:1062]Duplicate entry '1' for key 't.PRIMARY'
```

### PR DESCRIPTION
**Pull request by ekexium**
Signed-off-by: ekexium <eke@fastmail.com>

### What problem does this PR solve?

Issue Number: close #38242 

Problem Summary:

Print table name in error 1042: duplicate entry with the column/index name

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [x] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Improve the error message of 1042: duplicate entry, which makes it different from the error message in MySQL.
```

### CODE REVIEW

File: tests/realtikvtest/txntest/txn_test.go [hunk 1]
```diff
@@ -193,3 +193,22 @@ func TestWriteConflictReason(t *testing.T) {
...
+	tk.MustExec("set @@tx_isolation='read-committed'")
+	tk2.MustExec("use test")
+	tk.MustExec("set @@tidb_constraint_check_in_place_pessimistic=off")
+	tk.MustExec("create table t (c int primary key, v int)")
+	tk.MustExec("create table t2 (c int primary key, v int)")
+	tk.MustExec("begin pessimistic")
+	tk.MustExec("insert into t values (1, 1)")
+	tk2.MustExec("insert into t values (1, 1)")
+	tk2.MustExec("insert into t2 values (1, 2)")
+	_, err := tk.Exec("update t set v = v + 1 where c = 1")
```
Comment (hawkingrei) [inline tests/realtikvtest/txntest/txn_test.go:211]: Please use ```tk.MustContainErrMsg```.

**Comment by TonsnakeLin** [comment]
We should add a test case for unique-index I think.

**Comment by ekexium** [comment]
> We should add a test case for unique-index I think.

Good catch! I misses it.

**Comment by ekexium** [comment]
@TonsnakeLin @sticnarf  Could you help take a look at this and the corresponding mysql-PR? We're gonna merge them together.

**Comment by sticnarf** [comment]
LGTM but the CI seems unhappy...

**Review by sticnarf — APPROVED** [review]
LGTM

**Review by TonsnakeLin — APPROVED** [review]
No missing point of setting table name for ErrKeyExists is found. LGTM.

**Pull request merged**